tests/functional/src/main/java/org/mule/functional/api/classloading/isolation/AetherClassPathClassifier.java

```diff
@@ -0,0 +1,591 @@
+              && rootArtifact.getArtifactId().equals(pluginArtifact.getArtifactId())))
+          .collect(toList());
+    }
+
+    pluginsArtifacts.stream().forEach(pluginArtifact -> pluginUrlClassifications
+        .add(buildPluginUrlClassification(pluginArtifact, context, extensionPluginMetadataGenerator)));
+
+    extensionPluginMetadataGenerator.generateDslResources();
+    return pluginUrlClassifications;
+  }
+
+  private boolean isExtensionPlugin(Artifact artifact) {
+    return artifact.getExtension().equals(MULE_EXTENSION);
+  }
+
+  /**
+   * Classifies a plugin {@link Artifact}. {@value org.eclipse.aether.util.artifact.JavaScopes#COMPILE} dependencies will be
+   * resolved for building the {@link URL}'s for the class loader. For {@link Extension} annotated classes it will also generate
+   * its metadata.
+   *
+   * @param pluginArtifact {@link Artifact} that represents the plugin to be classified
+   * @param context {@link ClassPathClassifierContext} with settings for the classification process
+   * @param extensionPluginGenerator {@link ExtensionPluginMetadataGenerator} extensions metadata generator
+   * @return {@link PluginUrlClassification} for the plugin
+   */
+  private PluginUrlClassification buildPluginUrlClassification(Artifact pluginArtifact, ClassPathClassifierContext context,
+                                                               ExtensionPluginMetadataGenerator extensionPluginGenerator) {
+    List<URL> urls;
+    try {
+      List<Dependency> managedDependencies = dependencyResolver.readArtifactDescriptor(pluginArtifact).getManagedDependencies();
+
+      final DependencyFilter dependencyFilter = orFilter(classpathFilter(COMPILE),
+                                                         new PatternExclusionsDependencyFilter(context.getExcludedArtifacts()));
+      urls = toUrl(dependencyResolver.resolveDependencies(new Dependency(pluginArtifact, COMPILE),
+                                                          Collections.<Dependency>emptyList(), managedDependencies,
+                                                          dependencyFilter));
+    } catch (Exception e) {
+      throw new IllegalStateException("Couldn't resolve dependencies for plugin: '" + pluginArtifact + "' classification", e);
+    }
+
+    if (context.isExtensionMetadataGenerationEnabled()) {
+      File generatedMetadataFolder = extensionPluginGenerator.generateExtensionManifest(pluginArtifact, urls);
+      if (generatedMetadataFolder != null) {
+        URL generatedTestResources = toUrl(generatedMetadataFolder);
+
+        if (generatedTestResources != null) {
+          List<URL> appendedTestResources = newArrayList(generatedTestResources);
+          appendedTestResources.addAll(urls);
+          urls = appendedTestResources;
+        }
+      }
+    }
+    // TODO(gfernandes): MULE-10484 How could I check if exported classes belong to this plugin?
```

Comment:
Why do you want to do that? If a plugins says that it exports X, then the classloader will attempt to find X from the plugin's classloader. It will fail to do it, as expected, but that is not a problem in my opinion